webapi/src/main/scala/org/knora/webapi/slice/common/ValueTypes.scala

```diff
@@ -25,3 +29,37 @@ trait WithFrom[-I, +A] {
 }
 
 trait StringValueCompanion[A <: StringValue] extends WithFrom[String, A]
+trait IntValueCompanion[A <: IntValue]       extends WithFrom[Int, A]
+
+object ToValidation {
+  def validateOneWithFrom[A, B <: Value[?], E <: Throwable](
+    a: A,
+    validator: A => Either[String, B],
+    err: String => E
+  ): Validation[E, B] =
+    Validation.fromEither(validator(a)).mapError(err.apply)
+
+  /**
+   * Helper function to validate an optional value using a validator function.
+   *
+   * If the value is None, the validation will succeed with None.
+   *
+   * If the value is Some, the validation will use the validator function to validate the value.
+   * If the validation succeeds, the validated value will be wrapped in Some.
+   * If the validation fails, the error message will be mapped using the err function and the validation will fail.
+   *
+   * @param maybeA the optional value to validate
+   * @param validator the validator function
+   * @param err the error function to map the error message
+   *
+   * @return a Validation containing the validated value or a failed Validation containing the error constructed by err
+   */
+  def validateOptionWithFrom[A, B <: Value[?], E](
+    maybeA: Option[A],
+    validator: A => Either[String, B],
+    err: String => E
+  ): Validation[E, Option[B]] = maybeA match {
+    case Some(a) => Validation.fromEither(validator(a)).map(Some(_)).mapError(err.apply)
+    case None    => Validation.succeed(None)
+  }
```

Comment:
I generally like this, and it is definitely helpful as it is now. However, I think it could be somewhat simplified if we decided to change our use of validation from `Throwable` or some more specific Exception type, to `String` on the caller sides. This would spare us the generic `E` here.  
That is of course not for this PR, but how would you feel about this?

Reply:
Most of the times `E` was a subtype of `Throwable` or even `RequestRejectedException`.
I had a type constraint on `E` and even thought about a default value.

However, I decided to leave both out in order to make this little function more versatile and force the caller to decide for a specific `E`.